### PR TITLE
Host Endpoint converter and code cleanup for converter functions

### PR DESCRIPTION
**Pull request by mgleung**
I need to delete the `modelv2v1` files. Will do so if these changes look OK.

### CODE REVIEW

File: lib/backend/syncersv1/updateprocessors/hostendpointprocessor.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		EndpointID: v2res.GetName(),
+	}
+
+	var v1value *model.HostEndpoint
+	// Deletion operations will have empty values so skip if empty
+	if v2res != nil {
+		var ipv4Addrs []cnet.IP
+		var ipv6Addrs []cnet.IP
+		for _, ipString := range v2res.Spec.ExpectedIPs {
+			ip := *cnet.ParseIP(ipString)
```
Comment (robbrockbank) [inline lib/backend/syncersv1/updateprocessors/hostendpointprocessor.go]: probably need to check ip.IP != nil

File: lib/backend/syncersv1/updateprocessors/hostendpointprocessor.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	}
+
+	var v1value *model.HostEndpoint
+	// Deletion operations will have empty values so skip if empty
+	if v2res != nil {
+		var ipv4Addrs []cnet.IP
+		var ipv6Addrs []cnet.IP
+		for _, ipString := range v2res.Spec.ExpectedIPs {
+			ip := cnet.ParseIP(ipString)
+			if ip != nil {
```
Comment (robbrockbank) [inline lib/backend/syncersv1/updateprocessors/hostendpointprocessor.go:57]: if it is nil, it indicates a parsing error....

Comment (mgleung) [reply lib/backend/syncersv1/updateprocessors/hostendpointprocessor.go:57]: OK. I was thinking along the lines of the `ExpectedIPs` could accidentally have an empty string passed in which I wasn't sure if it qualified as an error.

Comment (robbrockbank) [reply lib/backend/syncersv1/updateprocessors/hostendpointprocessor.go:57]: yeah - i'll buy that!

**Comment by robbrockbank** [comment]
Looks great!

**Comment by mgleung** [comment]
Opened a different PR against libcalico-go since this branch was merged in there.  New PR is at https://github.com/projectcalico/libcalico-go/pull/569

**Pull request closed without merging**